File: backend/src/main/kotlin/com/getamotel/domain/property/Property.kt
package com.getamotel.domain.property

import com.getamotel.domain.TenantEntity
import jakarta.persistence.*
import java.time.LocalTime

@Entity
@Table(name = "properties")
class Property : TenantEntity() {

    @Column(nullable = false)
    lateinit var name: String

    @Column(nullable = false)
    lateinit var slug: String

    @Column(columnDefinition = "TEXT")
    var description: String? = null

    @Column(name = "address_line1", nullable = false)
    lateinit var addressLine1: String

    @Column(name = "address_line2")
    var addressLine2: String? = null

    @Column(nullable = false)
    lateinit var city: String

    @Column(nullable = false, length = 2)
    lateinit var state: String

    @Column(name = "zip_code", nullable = false, length = 10)
    lateinit var zipCode: String

    @Column(nullable = false, length = 2)
    var country: String = "US"

    var latitude: Double? = null
    var longitude: Double? = null

    var phone: String? = null
    var email: String? = null

    @Column(name = "check_in_time", nullable = false)
    var checkInTime: LocalTime = LocalTime.of(15, 0)

    @Column(name = "check_out_time", nullable = false)
    var checkOutTime: LocalTime = LocalTime.of(11, 0)

    @Column(name = "status", nullable = false)
    @Enumerated(EnumType.STRING)
    var status: PropertyStatus = PropertyStatus.ACTIVE
}

enum class PropertyStatus {
    ACTIVE, INACTIVE
}

File: frontend/lib/features/search/models/property.dart
import 'package:json_annotation/json_annotation.dart';

@JsonSerializable()
class Property {
  final String id;
  final String name;
  final String? description;
  final String addressLine1;
  final String city;
  final String state;
  final String zipCode;
  final String? phone;
  final String? email;

  Property({
    required this.id,
    required this.name,
    this.description,
    required this.addressLine1,
    required this.city,
    required this.state,
    required this.zipCode,
    this.phone,
    this.email,
  });

  factory Property.fromJson(Map<String, dynamic> json) => Property(
    id: json['id'],
    name: json['name'],
    description: json['description'],
    addressLine1: json['addressLine1'],
    city: json['city'],
    state: json['state'],
    zipCode: json['zipCode'],
    phone: json['phone'],
    email: json['email'],
  );
}

class PropertySearchResult {
  final Property property;
  final int lowestPriceCents;
  final int availableRoomTypeCount;

  PropertySearchResult({
    required this.property,
    required this.lowestPriceCents,
    required this.availableRoomTypeCount,
  });

  factory PropertySearchResult.fromJson(Map<String, dynamic> json) => PropertySearchResult(
    property: Property.fromJson(json['property']),
    lowestPriceCents: json['lowestPriceCents'],
    availableRoomTypeCount: json['availableRoomTypeCount'],
  );
}
